waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -67,16 +57,52 @@ genJobs spec = return $ genJob <$> getJobs spec
                   [ "jobName" .= jobName,
                     "jobPerformFnName" .= jobPerformFnName,
                     "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
-                    "jobFactoryName" .= show (jobFactoryForJob job)
+                    "jobPerformOptions" .= show (fromMaybe AS.JSON.emptyObject (J.options . J.perform $ job)),
+                    "executorJobRelFP" .= toFilePath (executorJobTemplateInJobsDir (J.executor job))
                   ]
             )
 
-genJobFactories :: Generator [FileDraft]
-genJobFactories = return $ genJobFactory <$> jobFactories
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+genJobExecutors :: Generator [FileDraft]
+genJobExecutors = return $ jobExecutorFds ++ jobExecutorHelperFds
   where
-    genJobFactory :: JobFactory -> FileDraft
-    genJobFactory jobFactory =
-      let jobFactoryFp = jobFactoryFilePath jobFactory
-          sourceTemplateFp = C.asTmplFile jobFactoryFp
-          destinationServerFp = C.asServerFile jobFactoryFp
-       in C.mkTmplFdWithDstAndData sourceTemplateFp destinationServerFp Nothing
+    jobExecutorFds :: [FileDraft]
+    jobExecutorFds = genJobExecutor <$> jobExecutors
+
+    genJobExecutor :: JobExecutor -> FileDraft
+    genJobExecutor jobExecutor = C.mkTmplFd $ executorJobTemplateInServerTemplatesDir jobExecutor
+
+    jobExecutorHelperFds :: [FileDraft]
+    jobExecutorHelperFds =
+      [ C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/pgBoss/pgBoss.js|],
+        C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/Job.js|],
+        C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/SubmittedJob.js|]
+      ]
+
+data JobsDir
```

Comment:
So this represents both Jobs dir in templates, and in generated project. 
Interesting.
I guess it makes sense!

Reply:
Yeah, I'm not 100% satisfied, but I think prob good enough for now. I can continue to think on it while wrapping up final phase. 👍🏻

Reply:
I agree it is good enough for now!